scripts/e2e/cmd/runner/helper.go

```diff
@@ -397,3 +434,45 @@ func makeGetRequest(url string, host string, statusCode int, inSecure bool) (*ht
 
 	return nil, fmt.Errorf("Unable to get status code %d with url '%s' with host '%s'. Response: [%+v]", statusCode, url, host, resp)
 }
+
+func validateSubResourcePrefix() error {
+	klog.Info("Validating that appgw sub resources are named with the prefix provided in helm")
+
+	env := GetEnv()
+	if env.SubResourceNamePrefix == "" {
+		return errors.New("value for SubResourceNamePrefix env variable is not set and so, can't validate that all appgw properties use the prefix")
+	}
+
+	klog.Info("preparing app gateway client")
+	client, err := getApplicationGatewaysClient()
+	if err != nil {
+		return err
+	}
+
+	gateway, err := client.Get(
+		context.TODO(),
+		env.ResourceGroupName,
+		env.AppGwName,
+	)
+
+	if err != nil {
+		return err
+	}
+
+	prefixUsed := false
+	for _, listener := range *gateway.HTTPListeners {
+		klog.Infof("checking listener %s for %s", *listener.Name, env.SubResourceNamePrefix)
+		if strings.HasPrefix(*listener.Name, env.SubResourceNamePrefix) {
+			klog.Infof("found %s that uses the prefix", *listener.Name)
+			prefixUsed = true
+			break
```

Comment:
what if other listener doesn't have the prefix? prefixUsed should be initialized to be true and check if anything doesn't have the prefix.

Reply:
since not all resources might use the prefix, just checking that prefix is getting to AGIC.
Cases where this would fail is when AppGW is shared or ssl certificates are pre-installed through portal.